### PR TITLE
stop the lease controller when cluster does not exist

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Delete cluster lease controller after cluster is deleted according to #590.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/status/cluster_status_controller.go [hunk 1]
```diff
@@ -244,7 +244,7 @@ func (c *ClusterStatusController) buildInformerForCluster(cluster *v1alpha1.Clus
 	return singleClusterInformerManager, nil
 }
 
-func (c *ClusterStatusController) initLeaseController(cluster *v1alpha1.Cluster) {
+func (c *ClusterStatusController) initLeaseController(cluster *v1alpha1.Cluster, stopCh <-chan struct{}) {
```
Comment (RainbowMango) [inline pkg/controllers/status/cluster_status_controller.go:247]: Would it be better to take `context` as a parameter than a `chan`?

Comment (Garrybest) [reply pkg/controllers/status/cluster_status_controller.go:247]: Good idea.

**Review by RainbowMango — APPROVED** [review]
/lgtm

Have you tested it manually? I'm asking because we don't have a test cover it.

**Comment by Garrybest** [comment]
Right. I have tested this PR and found the lease would be deleted normally when called `kubectl delete cluster xxx`. No error has occured.

**Comment by RainbowMango** [comment]
Thanks. Let's go!

**Pull request merged**